py3status/modules/mpris.py

```diff
@@ -375,16 +375,21 @@ def _add_player(self, player_id):
         """
         Add player to mpris_players
         """
+        if not player_id.startswith(SERVICE_BUS):
+            return False
+
         try:
             player = self._dbus.get(player_id, SERVICE_BUS_URL)
-            if player_id.startswith(SERVICE_BUS):
-                if player.Identity not in self._mpris_names:
-                    self._mpris_names[player.Identity] = player_id.split('.')[-1]
-                    for p in self._mpris_players.values():
-                        if not p['name'] and p['identity'] in self._mpris_names:
-                            p['name'] = self._mpris_names[p['identity']]
-                            p['full_name'] = u'{} {}'.format(p['name'], p['index'])
+
+            if player.Identity in self._mpris_names:
```

Comment:
I don't think this is quite right. when we remove a player then we don't remove it from `_mpris_names`

for me if I restart py3status with no media player.  If I then start vlc (nothing playing) it works correctly.  When I close vlc all is good.  but if I then start vlc it is not registered.

for me  `if player.Identity in self._mpris_players:` works better. but still does not loctae the player about one time in three.

I'll do some more playing.

I also do not have a consistent test case for #707 to see if that specific issue is fix as the reports suggest